src/feature-test/java/uk/gov/ida/verifyserviceprovider/configuration/ApplicationConfigurationFeatureTests.java

```diff
@@ -0,0 +1,129 @@
+            put("MSA_ENTITY_ID", "some-msa-entity-id");
+            put("SERVICE_ENTITY_IDS", "[\"http://some-service-entity-id\"]");
+            put("HASHING_ENTITY_ID", "some-hashing-entity-id");
+            put("SAML_SIGNING_KEY", TEST_RP_PRIVATE_SIGNING_KEY);
+            put("SAML_PRIMARY_ENCRYPTION_KEY", TEST_RP_PRIVATE_ENCRYPTION_KEY);
+            put("SAML_SECONDARY_ENCRYPTION_KEY", TEST_RP_PRIVATE_ENCRYPTION_KEY);
+            put("CLOCK_SKEW", "PT30s");
+        }}.forEach(environmentVariables::set);
+    }
+
+    @After
+    public void cleanup() {
+        application.getTestSupport().after();
+    }
+
+    @Test
+    public void applicationShouldStartUp() throws Exception {
+        application.getTestSupport().before();
+
+        VerifyServiceProviderConfiguration configuration = application.getConfiguration();
+
+        assertThat(application.getLocalPort()).isEqualTo(50555);
+        assertThat(((DefaultLoggingFactory) configuration.getLoggingFactory()).getLevel()).isEqualTo("ERROR");
+        assertThat(configuration.getHubSsoLocation().toString()).isEqualTo(HubEnvironment.COMPLIANCE_TOOL.getSsoLocation().toString());
+        assertThat(configuration.getHubMetadataConfiguration().getUri().toString()).isEqualTo("https://compliance-tool-reference.ida.digital.cabinet-office.gov.uk/SAML2/metadata/federation");
+        assertThat(configuration.getHubMetadataConfiguration().getExpectedEntityId()).isEqualTo("https://signin.service.gov.uk");
+        assertThat(configuration.getMsaMetadataConfiguration().getExpectedEntityId()).isEqualTo("some-msa-entity-id");
+        assertThat(configuration.getMsaMetadataConfiguration().getUri().toString()).isEqualTo("some-msa-metadata-url");
+        assertThat(configuration.getServiceEntityIds()).containsExactly("http://some-service-entity-id");
+        assertThat(configuration.getSamlSigningKey().getEncoded()).isEqualTo(decode(TEST_RP_PRIVATE_SIGNING_KEY));
+        assertThat(configuration.getSamlPrimaryEncryptionKey().getEncoded()).isEqualTo(decode(TEST_RP_PRIVATE_ENCRYPTION_KEY));
+        assertThat(configuration.getSamlSecondaryEncryptionKey().getEncoded()).isEqualTo(decode(TEST_RP_PRIVATE_ENCRYPTION_KEY));
+        assertThat(configuration.getClockSkew()).isEqualTo(Duration.standardSeconds(30));
+    }
+
+    @Test
+    public void applicationShouldStartUpWithListOfServiceEntityIds() throws NoSuchFieldException, IllegalAccessException {
+        environmentVariables.set("SERVICE_ENTITY_IDS", "[\"http://some-service-entity-id\",\"http://some-other-service-entity-id\"]");
+        application.getTestSupport().before();
+
+        VerifyServiceProviderConfiguration configuration = application.getConfiguration();
+
+        assertThat(configuration.getServiceEntityIds()).containsExactly("http://some-service-entity-id", "http://some-other-service-entity-id");
+        assertThat(configuration.getHashingEntityId()).isEqualTo("some-hashing-entity-id");
```

Comment:
Can we replace the hardcoded `some-hashing-entity-id` string with the constant variable?